demos/ios/MASVS-RESILIENCE/MASTG-DEMO-0023/MastgTest.swift

```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+import MachO
+
+struct MastgTest {
+    static func checkDebuggerEntitlement() -> Bool {
+        print("\n=== Starting Debug Check ===\n")
+        
+        // Method 1: Check Bundle Info
+        print("Method 1: Checking Bundle Info")
+        let bundleURL = Bundle.main.bundleURL
+        print("📦 Bundle URL: \(bundleURL.path)")
+        
+        // Try to read Info.plist
+        let infoPath = bundleURL.appendingPathComponent("Info.plist").path
+        print("📄 Info.plist Path: \(infoPath)")
+        
+        if let infoDict = NSDictionary(contentsOfFile: infoPath) {
+            print("📋 Info.plist Contents: \(infoDict)")
+        }
+        
+        // Method 2: Check Executable
+        print("\nMethod 2: Checking Executable")
+        if let executablePath = Bundle.main.executablePath {
+            print("📍 Executable Path: \(executablePath)")
+            
+            do {
+                let fileManager = FileManager.default
+                let attributes = try fileManager.attributesOfItem(atPath: executablePath)
+                print("📊 File Attributes: \(attributes)")
+                
+                // Try reading with different methods
+                let data = try Data(contentsOf: URL(fileURLWithPath: executablePath))
+                print("📄 File Size: \(data.count) bytes")
+                
+                if data.count > 0 {
+                    if let content = String(data: data, encoding: .ascii),
+                       content.contains("get-task-allow") {
+                        print("✅ Found get-task-allow in binary!")
+                        return true
+                    }
+                }
+            } catch {
+                print("❌ Error reading executable: \(error)")
+            }
+        }
+        
+        // Method 3: Check Embedded.mobileprovision
+        print("\nMethod 3: Checking Embedded.mobileprovision")
+        let bundlePath = Bundle.main.bundlePath
+        let provisionPath = (bundlePath as NSString).appendingPathComponent("embedded.mobileprovision")
+        print("📄 Provision Path: \(provisionPath)")
+        
+        do {
+            let provisionData = try Data(contentsOf: URL(fileURLWithPath: provisionPath))
+            if let provisionString = String(data: provisionData, encoding: .ascii),
+               provisionString.contains("get-task-allow") {
+                print("✅ Found get-task-allow in provision profile!")
+                return true
+            }
+            print("📄 Provision Size: \(provisionData.count) bytes")
+        } catch {
+            print("⚠️ No embedded provision profile found: \(error)")
+        }
+        
+        // Method 4: Process Info
+        print("\nMethod 4: Checking Process Info")
+        let isDebugger = isDebuggerAttached()
```

Comment:
Can you make such a function for each method? It would clean up the code a bit.

Reply:
Do we need this file at all? If I'm not wrong I think we just need the modified entitlements file, so that we can build the app using it, and the reversed version.

Here's the original: https://github.com/cpholguera/MASTestApp-iOS/blob/main/entitlements.plist

Reply:
Reducing this demo to only one method is also ok